src/lib/import_export/binary/binary_writer.hpp

```diff
@@ -125,12 +125,35 @@ class BinaryWriter {
    * ^: These fields are only written if the type of the column IS a string.
    * °: This field is written if the type of the column is NOT a string
    *
-   * @param base_dictionary_segment The segment to export
+   * @param dictionary_segment The segment to export
    * @param ofstream The output stream for exporting
    */
   template <typename T>
   static void _write_segment(const DictionarySegment<T>& dictionary_segment, std::ofstream& ofstream);
 
+  /**
+   * FixedStringDictionary Segments are dumped with the following layout:
+   *
+   * Description           | Type                                  | Size in bytes
+   * -----------------------------------------------------------------------------------------
+   * Column Type           | ColumnType                            |   1
+   * Width of attribute v. | AttributeVectorWidth                  |   1
+   * Size of dictionary v. | ValueID                               |   4
+   * FixedString length    | uint32_t                              |   8
+   * Dictionary Values     | char[]                                |   dict. size * fixed string length
+   * Attribute v. values   | uintX                                 |   rows * width of attribute v.
+   *
+   * Please note that the number of rows are written in the header of the chunk.
+   * The type of the column can be found in the global header of the file.
```

Comment:
This is a good comment. I actually looked at other code first, and had that exact question.

This applies to every encoding, right? It might not hurt to copy it.

Reply:
Actually, I copied it from DictionarySegment comment; this is part of every segment export comment.